File: c_sharp/Enumeration/Enumeration/Gender.cs
using PeerageHRIS.Resources;
using System.ComponentModel.DataAnnotations;

namespace PeerageHRIS.Enumeration
{
    public enum Gender
    {
        [Display(Name = "Male", ResourceType = typeof(CommonResources))]
        Male,
        [Display(Name = "Female", ResourceType = typeof(CommonResources))]
        Female,
        [Display(Name = "Other", ResourceType = typeof(CommonResources))]
        Other
    }
}

File: lib/core/enums/gender.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'gender.g.dart';

@JsonEnum(alwaysCreate: true)
enum Gender {
  @JsonValue(0)
  male,
  @JsonValue(1)
  female,
  @JsonValue(2)
  other,
}
